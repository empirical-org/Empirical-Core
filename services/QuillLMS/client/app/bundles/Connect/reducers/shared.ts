
export function receiveQuestionsData(currentstate, action) {
  if (action.language) {
    return Object.assign({}, currentstate, {
      translated_data: {
        ...currentstate.translated_data,
        [action.language] : action.data
      },
    });
  } else {
    return Object.assign({}, currentstate, {
      hasreceiveddata: true,
      data: action.data,
    });
  }
}

export function receiveQuestionData(currentstate, action) {
  return Object.assign({}, currentstate, {
    data: Object.assign({}, currentstate.data, {
      [action.uid]: action.data,
    })
  })
}
