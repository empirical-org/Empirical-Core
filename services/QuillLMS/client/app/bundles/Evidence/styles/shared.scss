.evidence-step-number-small {
  width: 100%;
  border-radius: 16px;
  justify-content: center;
  align-items: center;
  display: flex;
  font-size: 16px;
  font-weight: 600;
  margin-right: 8px;
  background-color: #005341;
  color: #ffffff;
  max-width: 24px;
  height: 24px;
  font-size: 14px;
  &.active {
    background-color: #ffffff;
    color: #262626;
  }
}
