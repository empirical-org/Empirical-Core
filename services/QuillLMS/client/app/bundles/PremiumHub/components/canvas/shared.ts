export interface CanvasInstance {
  client_id: string;
  id: number;
  client_secret: string;
  school_ids: number[];
  school_names: string[];
  url: string;
}

export interface School {
  id: number;
  name: string;
  subscription_status: any;
  subscriptions: any[];
}
