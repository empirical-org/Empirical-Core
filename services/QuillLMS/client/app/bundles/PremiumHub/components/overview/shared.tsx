export const USAGE_HIGHLIGHTS = 'Usage Highlights'
export const PROFESSIONAL_DEVELOPMENT_AND_SUPPORT = 'Professional Development and Support'
export const ACCOUNT_MANAGEMENT = 'Account Management'
export const PREMIUM_REPORTS = 'Premium Reports'
export const INTEGRATIONS = 'Integrations'

export const iconLinkBase = `${process.env.CDN_URL}/images/pages/administrator/overview`

export const SECTION_NAME_TO_ICON_URL = {
  [USAGE_HIGHLIGHTS]: `${iconLinkBase}/bulb.svg`,
  [PROFESSIONAL_DEVELOPMENT_AND_SUPPORT]: `${iconLinkBase}/students.svg`,
  [ACCOUNT_MANAGEMENT]: `${iconLinkBase}/pencil.svg`,
  [PREMIUM_REPORTS]: `${iconLinkBase}/bar-graph-increasing-black.svg`,
  [INTEGRATIONS]: `${iconLinkBase}/checkbox-multiple.svg`
}
