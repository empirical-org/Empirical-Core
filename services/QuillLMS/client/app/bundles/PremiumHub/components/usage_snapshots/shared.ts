export const SMALL = 'small'
export const MEDIUM = 'medium'
export const POSITIVE = 'positive'
export const NEGATIVE = 'negative'
