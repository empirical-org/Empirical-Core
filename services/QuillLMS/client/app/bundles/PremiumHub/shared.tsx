import * as React from 'react'
import * as _ from 'lodash'

import * as md5 from 'md5'

import { unorderedArraysAreEqual, } from '../../modules/unorderedArraysAreEqual'

export const RESTRICTED = 'restricted'
export const LIMITED = 'limited'
export const FULL = 'full'
export const LOADING = 'loading'

export const APPROVED = 'Approved'
export const PENDING = 'Pending'
export const DENIED = 'Denied'
export const SKIPPED = 'Skipped'

export const OVERVIEW = 'overview'
export const SKILL = 'skill'
export const STUDENT = 'student'

export const DIAGNOSTIC_REPORT_DEFAULT_CELL_WIDTH = '182px'
export const groupByDropdownOptions = [{ label: 'Grade', value: 'grade' }, { label: 'Teacher', value: 'teacher' }, { label: 'Classroom', value: 'classroom' }]
export const diagnosticTypeDropdownOptions = [
  { label: 'Starter Baseline Diagnostic (Pre)', value: 'Starter Baseline Diagnostic (Pre)' },
  { label: 'Starter Growth Diagnostic (Post)', value: 'Starter Growth Diagnostic (Post)' },
  { label: 'Intermediate Baseline Diagnostic (Pre)', value: 'Intermediate Baseline Diagnostic (Pre)' },
  { label: 'Intermediate Growth Diagnostic (Post)', value: 'Intermediate Growth Diagnostic (Post)' },
  { label: 'Advanced Baseline Diagnostic (Pre)', value: 'Advanced Baseline Diagnostic (Pre)' },
  { label: 'Advanced Growth Diagnostic (Post)', value: 'Advanced Growth Diagnostic (Post)' },
  { label: 'ELL Starter Baseline Diagnostic (Pre)', value: 'ELL Starter Baseline Diagnostic (Pre)' },
  { label: 'ELL Starter Growth Diagnostic (Post)', value: 'ELL Starter Growth Diagnostic (Post)' },
  { label: 'ELL Intermediate Baseline Diagnostic (Pre)', value: 'ELL Intermediate Baseline Diagnostic (Pre)' },
  { label: 'ELL Intermediate Growth Diagnostic (Post)', value: 'ELL Intermediate Growth Diagnostic (Post)' },
  { label: 'ELL Advanced Baseline Diagnostic (Pre)', value: 'ELL Advanced Baseline Diagnostic (Pre)' },
  { label: 'ELL Advanced Growth Diagnostic (Post)', value: 'ELL Advanced Growth Diagnostic (Post)' },
  { label: 'AP Writing Skills Survey', value: 'AP Writing Skills Survey' },
  { label: 'Pre-AP Writing Skills Survey 1', value: 'Pre-AP Writing Skills Survey 1' },
  { label: 'Pre-AP Writing Skills Survey 2', value: 'Pre-AP Writing Skills Survey 2' },
  { label: 'SpringBoard Writing Skills Survey', value: 'SpringBoard Writing Skills Survey' }
]

export const premiumLockImage = <img alt="Gray lock" src={`${process.env.CDN_URL}/images/pages/administrator/premium_lock.svg`} />

export const restrictedElement = (
  <div className="restricted">
    {premiumLockImage}
  </div>
)

export const restrictedPage = (
  <div className="restricted-page">
    {premiumLockImage}
  </div>
)

export function selectionsEqual(firstSelection, secondSelection) {
  return (
    firstSelection == secondSelection || // less strict comparison so that undefined and null are treated as equal
    unorderedArraysAreEqual(firstSelection, secondSelection)
  )
}

export function mapItemsIfNotAll(selectedItems, allItems, mapKey = 'id') {
  // selectedItems may, by design, be a superset of allItems, but if everything in allItems is in selectedItems, we want to treat it as "everything" being selected
  const allItemsSelected = allItems.every((item) => {
    return _.some(selectedItems, item)
  })

  if (allItemsSelected || selectionsEqual(selectedItems, allItems)) return null

  return selectedItems.map(i => i[mapKey])
}

/* Should produce the same output as backend `app/services/payload_hasher.rb:run` */
export function hashPayload(payloadArray: Array<any>) {
  const joinedPayload = payloadArray.join('-')
  return md5(joinedPayload)
}
