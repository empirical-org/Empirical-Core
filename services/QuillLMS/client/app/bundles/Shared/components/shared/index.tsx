export {
  ArchivedButton
} from './archivedButton'

export {
  PostNavigationBanner
} from './postNavigationBanner'

export {
  ButtonLoadingSpinner,
  DarkButtonLoadingSpinner,
  LightButtonLoadingSpinner,
} from './buttonLoadingSpinner'

export {
  Card
} from './card'

export {
  Checkbox
} from './checkBox'

export {
  DragHandle
} from './dragHandle'

export {
  ExpandableCard
} from './expandableCard'

export {
  CarouselAnimation
} from './carouselAnimation'

export {
  DataTable
} from './dataTable'

export {
  ProgressBar
} from './progressBar'

export {
  DropdownInput
} from './dropdownInput'

export {
  DropdownInputWithSearchTokens
} from './dropdownInputWithSearchTokens'

export {
  Error
} from './error'

export {
  FlagDropdown
} from './flagDropdown'

export {
  HelpfulTips
} from './helpfulTips'

export {
  Input
} from './input'

export {
  Passthrough
} from './passthrough'

export {
  ResumeOrBeginButton
} from './resumeOrBeginButton'

export {
  ReportHeader
} from './reportHeader'

export {
  ScreenreaderInstructions
} from './screenreaderInstructions'

export {
  OneThumbSlider,
  TwoThumbSlider,
} from './slider'

export {
  SmartSpinner
} from './smartSpinner'

export {
  Snackbar,
  defaultSnackbarTimeout
} from './snackbar'

export {
  SortableList
} from './sortableList'

export {
  Spinner
} from './spinner'

export {
  TeacherPreviewMenu
} from './teacherPreviewMenu'

export {
  TeacherPreviewMenuButton
} from './teacherPreviewMenuButton'

export {
  TextEditor
} from './textEditor'

export {
  TextArea
} from './textArea'

export {
  Tooltip
} from './tooltip'

export {
  UploadOptimalResponses
} from './uploadOptimalResponses'

export {
  ToggleComponentSection
} from './toggleComponentSection'
