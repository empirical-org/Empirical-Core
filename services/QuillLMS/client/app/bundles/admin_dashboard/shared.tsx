import * as React from 'react'

export const RESTRICTED = 'restricted'
export const LIMITED = 'limited'
export const FULL = 'full'

export const premiumLockImage = <img alt="Gray lock" src={`${process.env.CDN_URL}/images/pages/administrator/premium_lock.svg`} />

export const restrictedElement = (
  <div className="restricted">
    {premiumLockImage}
  </div>
)
