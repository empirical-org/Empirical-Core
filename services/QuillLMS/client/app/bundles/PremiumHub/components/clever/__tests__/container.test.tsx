import * as React from "react" ;
import { render, }  from "@testing-library/react";

import CleverIntegrationContainer from '../container'
import { FULL, RESTRICTED, } from '../../../shared'

describe('CleverIntegrationContainer', () => {

  describe('when access is restricted', () => {
    test('it should render', () => {
      const { asFragment } = render(<CleverIntegrationContainer accessType={RESTRICTED} />);
      expect(asFragment()).toMatchSnapshot();
    })
  })

  describe('when access is full', () => {
    test('it should render', () => {
      const { asFragment } = render(<CleverIntegrationContainer accessType={FULL} />);
      expect(asFragment()).toMatchSnapshot();
    })
  })

})
