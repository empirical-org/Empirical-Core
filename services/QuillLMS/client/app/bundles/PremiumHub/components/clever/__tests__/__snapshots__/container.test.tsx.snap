// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CleverIntegrationContainer when access is full it should render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="integration-container"
    >
      <img
        alt=""
        class="logo"
        src="undefined/images/pages/administrator/integrations/clever_logo.svg"
      />
      <h1>
        Streamline teaching with Clever
      </h1>
      <p>
        Make teaching at your school even more effective with our seamless Clever integration. With this integration, teachers at your school or district can:
      </p>
      <ul>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Seamlessly import Clever rosters.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Automatically create and sync Clever student accounts.
        </li>
      </ul>
      <p>
        You can also access Clever Secure Sync as part of your subscription. This makes the integration even more powerful and easy to configure.
      </p>
      <div
        class="links"
      >
        <a
          href="https://support.quill.org/en/articles/8500172-how-to-choose-your-rostering-integration"
          rel="noopener noreferrer"
          target="_blank"
        >
          How to choose your integration
        </a>
        <a
          href="https://support.quill.org/en/articles/8461682-how-do-i-set-up-clever-secure-sync-for-my-school-district-for-clever-quill-administrators"
          rel="noopener noreferrer"
          target="_blank"
        >
          How to set up this integration
        </a>
      </div>
      <div
        class="integration-tip"
      >
        <div>
          <h2>
            Integration Tip
          </h2>
          <p>
            Start the school year right by choosing a single integration. Changing mid-year can cause rostering challenges and may take up more of your valuable time, so decide upfront for a better experience.
          </p>
        </div>
        <img
          alt=""
          src="undefined/images/pages/administrator/integrations/lightbulb.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CleverIntegrationContainer when access is restricted it should render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="integration-container"
    >
      <img
        alt=""
        class="logo"
        src="undefined/images/pages/administrator/integrations/clever_logo.svg"
      />
      <h1>
        Streamline teaching with Clever
      </h1>
      <p>
        Make teaching at your school even more effective with our seamless Clever integration. With this integration, teachers at your school or district can:
      </p>
      <ul>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Seamlessly import Clever rosters.
        </li>
        <li>
          <img
            alt=""
            src="undefined/images/pages/administrator/integrations/circle-check.svg"
          />
          Automatically create and sync Clever student accounts.
        </li>
      </ul>
      <p>
        You can also subscribe to School or District Premium to access Clever Secure Sync. This makes the integration even more powerful and easy to configure.
      </p>
      <div
        class="links"
      >
        <a
          href="/premium"
          rel="noopener noreferrer"
          target="_blank"
        >
          Explore premium
        </a>
        <a
          href="https://support.quill.org/en/articles/8500172-how-to-choose-your-rostering-integration"
          rel="noopener noreferrer"
          target="_blank"
        >
          How to choose your integration
        </a>
        <a
          href="https://support.quill.org/en/articles/8461682-how-do-i-set-up-clever-secure-sync-for-my-school-district-for-clever-quill-administrators"
          rel="noopener noreferrer"
          target="_blank"
        >
          How to set up this integration
        </a>
      </div>
      <div
        class="integration-tip"
      >
        <div>
          <h2>
            Integration Tip
          </h2>
          <p>
            Start the school year right by choosing a single integration. Changing mid-year can cause rostering challenges and may take up more of your valuable time, so decide upfront for a better experience.
          </p>
        </div>
        <img
          alt=""
          src="undefined/images/pages/administrator/integrations/lightbulb.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
